airbyte-workers/src/main/java/io/airbyte/workers/helper/FailureHelper.java

```diff
@@ -46,6 +47,14 @@ public static FailureReason destinationFailure(final Throwable t, final Long job
         .withExternalMessage("Something went wrong within the destination connector");
   }
 
+  public static FailureReason checkFailure(final Throwable t, final Long jobId, final Integer attemptNumber, FailureReason.FailureOrigin origin) {
+    return genericFailure(t, jobId, attemptNumber)
+        .withFailureOrigin(origin)
+        .withFailureType(FailureReason.FailureType.CONFIG_ERROR)
+        .withRetryable(false)
+        .withExternalMessage(String.format("%s check failed", origin));
+  }
```

Comment:
Here's a new FailureHelper we can use for the checks - I had guessed `.withRetryable(false)` would prevent retries, but that doesn't seem to be the case.